lib/cache/ets.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Provider.Cache.ETS do
+  @moduledoc """
+  An ets based cache implementation
+  """
+  @behaviour Provider.Cache
+
+  use GenServer
+
+  @spec start_link(Keyword.t()) :: GenServer.server()
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
+
+  @impl Provider.Cache
+  def set(module, key, value) do
+    GenServer.call(__MODULE__, {:set, module, key, value})
```

Comment:
Don't think this needs to go through the GenServer.

Reply:
Unless my understanding is incorrect, we would need to enable write concurrency to not put this through the GenServer. Though, ideally, only a single process would ever update this anyways, so maybe it is not an issue? But I think this issue also disappears with your other comment of making the config module also a GenServer.

Reply:
According to [docs](https://www.erlang.org/doc/apps/stdlib/ets.html#insert/2):

> The entire operation is guaranteed to be [atomic and isolated](https://www.erlang.org/doc/apps/stdlib/ets#module-concurrency), even when a list of objects is inserted.

So you're safe running this from multiple processes. I don't think write concurrency is needed here, because there won't be multiple simultaneous writes of different rows. We always store everything, and that's done with a single atomic & isolated ETS operation.

Reply:
Needed to make the table `:public` in order for that to work.

Reply:
Yeah, forgot about that. I think that's fine.